test/jdk/java/nio/file/attribute/BasicFileAttributeView/CreationTime.java

```diff
@@ -106,8 +106,13 @@ static void test(Path top) throws IOException {
                 supportsCreationTimeWrite = true;
             }
         } else if (Platform.isLinux()) {
-            // Creation time read depends on statx system call support
-            supportsCreationTimeRead = Linker.nativeLinker().defaultLookup().find("statx").isPresent();
+            // Creation time read depends on statx system call support and on the file
+            // system storing the birth time. The tmpfs file system type does not store
+            // the birth time.
+            boolean statxIsPresent = Linker.nativeLinker().defaultLookup().find("statx").isPresent();
+            if (statxIsPresent && !Files.getFileStore(file).type().contentEquals("tmpfs")) {
+                supportsCreationTimeRead = true;
+            }
```

Comment:
Noting here that the test could still fail on some other filesystem which might not support birth time. For example `nfs` on Linux (IIRC).

Reply:
It is difficult to enumerate which file types in Linux do and do not support creation time.
How about add a function that use `stat -c` linux command line to determise support or not.

+    /**
+     * read the output of linux command `stat -c "%w" file`, if the output is "-",
+     * then the file system doesn't support birth time
+     */
+    public static boolean supportBirthTimeOnLinux(Path file) {
+        try {
+            String filePath = file.toAbsolutePath().toString();
+            ProcessBuilder pb = new ProcessBuilder("stat", "-c", "%w", filePath);
+            pb.redirectErrorStream(true);
+            Process p = pb.start();
+            BufferedReader b = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String l = b.readLine();
+            if (l != null && l.equals("-")) { return false; }
+        } catch(Exception e) {
+        }
+        return true;
+    }

Reply:
It's a possibility. I'm fine with the current "tmpfs" special case in the test, but I guess what you proposed above would be more robust. So if others are OK with such an approach, that would be fine with me as well.